Source/MQTTnet/Server/MqttSubscriptionInterceptorContext.cs

```diff
@@ -1,17 +1,19 @@
-﻿using System;
+﻿using MQTTnet.Packets;
+using System;
 
 namespace MQTTnet.Server
 {
     public class MqttSubscriptionInterceptorContext
     {
-        public MqttSubscriptionInterceptorContext(string clientId, TopicFilter topicFilter)
+        public MqttSubscriptionInterceptorContext(MqttConnectPacket connectPacket, TopicFilter topicFilter)
         {
-            ClientId = clientId;
+            ClientId = connectPacket.ClientId;
+            ConnectPacket = connectPacket;
             TopicFilter = topicFilter ?? throw new ArgumentNullException(nameof(topicFilter));
         }
 
         public string ClientId { get; }
-
+        public MqttConnectPacket ConnectPacket { get; }
```

Comment:
okay so what Properties should be captured of the connect packet? Should we place them in a new class or just pass them through seperately?

Reply:
I am not sure but basically I would not add things like protocol version. Same applies for properties wher WillDelayInterval is maybe not that relevant. I would to security relevant stuff only in the first place. And I would add  a new property for each new piece of information. Then the concept of properties in MQTTv5 is hidden. Also some properties must be nullable (like the original ones in the packet properties due to MQTT 3.1.1 support). 

Another thing is that the password should be protected in memory. Now we keep the password in RAM all the time which can be exposed. Maybe it is a good idea to put that in a "SecureString" and only reading in then required (like in Python scripts). But unfortunately this requires a password as a string but in MQTT it is a byte array...

Reply:
And you can also add the CONNECT packet to the constructor and just add getter properties to return the value from the private CONNECT packet. This will avoid creating lots of copies of the actual values and also avoid and endless constructor.

Reply:
i agree with your proposal, just to make it clear. the connect packet was stored in the connection all along so the password was in memory all the time.

Reply:
Oh was/is it? Then we also have think about that as well (later). I also want to add a "Items" property of type ConcurrentDictionary (similar to ASP.NET Session Items) to the session and expose that for all interceptors etc. This would allow me to implement some things much easier.